src/poetry/core/packages/url_dependency.py

```diff
@@ -11,11 +11,13 @@ def __init__(
         self,
         name: str,
         url: str,
+        directory: str | None = None,
```

Comment:
```suggestion
        *,
        directory: str | None = None,
```
I'm wavering whether we should add the new argument at the end to avoid breakage for potential downstream users or if we should make it clean. I tend to clean it up because I don't know of any downstream users.